qgis/Dockerfile

```diff
@@ -1,38 +1,59 @@
+# SHELL ["/bin/bash", "-c"]
+# RUN echo 'Types: deb deb-src' > /etc/apt/sources.list.d/qgis.sources
+# RUN echo 'URIs: https://qgis.org/ubuntugis' >> /etc/apt/sources.list.d/qgis.sources
+# RUN echo 'Suites: jammy' >> /etc/apt/sources.list.d/qgis.sources
+# RUN echo 'Architectures: amd64' >> /etc/apt/sources.list.d/qgis.sources
+# RUN echo 'Components: main' >> /etc/apt/sources.list.d/qgis.sources
+# RUN echo 'Signed-By: /etc/apt/keyrings/qgis-archive-keyring.gpg' >> /etc/apt/sources.list.d/qgis.sources
+# RUN apt-get update
 # here using apt-get leads 10 packages kept back which in turn prevents the successful installation of qgis, ok, use --with-new-pkgs!
-RUN apt-get -y --with-new-pkgs upgrade && \
-  apt-get -y autoremove && \
-  apt-get -y install qgis qgis-plugin-grass saga
+# RUN apt-get -y --with-new-pkgs upgrade && \
+#   apt-get -y autoremove && \
+#   apt-get -y install qgis qgis-plugin-grass saga
 
 # for how to use the qgis-plugin-manager, see https://github.com/3liz/qgis-plugin-manager
-RUN pip3 install qgis-plugin-manager
+# RUN apt install pip3
+# RUN pip3 install qgis-plugin-manager
 # to enable the qgis-plugin-manager, add the corresponding path to PATH
 ENV PATH="/home/rstudio/.local/bin:$PATH"
 # from the next line onwards we have trouble with the rstudio server, therefore we switch to the rstudio user
 USER rstudio
 RUN mkdir -p /home/rstudio/.local/share/QGIS/QGIS3/profiles/default/python/plugins
 ENV QGIS_PLUGINPATH=/home/rstudio/.local/share/QGIS/QGIS3/profiles/default/python/plugins
-RUN qgis-plugin-manager init
-RUN qgis-plugin-manager update
+
+# install QGIS plugins ################################################################################################
+# Using qgisprocess workflow:
+# See https://github.com/r-spatial/qgisprocess/blob/main/.github/workflows/R-CMD-check.yaml#L64
+# RUN echo "QGIS_PLUGINPATH=$(pwd)/extra_plugins" >> $GITHUB_ENV
+# This bit, with user as root, failed in Docker image, changing to rstudio:
+USER root
+RUN mkdir -p /root/.gnupg
+RUN chmod 700 /root/.gnupg
+# Version that works in image:
+RUN gpg --no-default-keyring --keyring /etc/apt/keyrings/ubuntugis-unstable-archive-keyring.gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 6B827C12C2D425E227EDCA75089EBE08314DF160
+RUN sh -c 'echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/ubuntugis-unstable-archive-keyring.gpg] http://ppa.launchpad.net/ubuntugis/ubuntugis-unstable/ubuntu `lsb_release -c -s` main" > /etc/apt/sources.list.d/ubuntugis-unstable.list'
+RUN wget -qO /etc/apt/keyrings/qgis-archive-keyring.gpg https://download.qgis.org/downloads/qgis-archive-keyring.gpg
+RUN sh -c 'echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/qgis-archive-keyring.gpg] https://qgis.org/ubuntugis `lsb_release -c -s` main" > /etc/apt/sources.list.d/qgis.list'
+RUN apt-get update
+RUN apt-get install -y qgis qgis-plugin-grass saga
+RUN wget -qO sagang_plugin.zip https://plugins.qgis.org/plugins/processing_saga_nextgen/version/0.0.7/download/
```

Comment:
just curious, what is the advantage of installing the saga plugin manually? Is it because of having one dependency (qgis-plugin-manager) less? Or because the qgis-plugin-manager is probably only available for Linux?

Reply:
> just curious, what is the advantage of installing the saga plugin manually? Is it because of having one dependency (qgis-plugin-manager) less? Or because the qgis-plugin-manager is probably only available for Linux?

2 reasons come mind:

- The previous approach, which included a tool to manage plugins with `pip3 install ...` command did not work, and the Python package does not seem to be widely used or well-maintained.
- We want tight integration with the **`qgisprocess`** so I followed their lead, as outlined in the message explaining this: 

https://github.com/geocompx/docker/blob/681dfe45815c99543243a0d74697acfdb14d3650/qgis/Dockerfile#L36

I'm not sure why they used this. Looking at the Blame it seems that line was added by @florisvdh in this commit: https://github.com/r-spatial/qgisprocess/commit/ebca4f1e20b63689d936b7c6c972a5004c7bb314

I'd be grateful for any feedback here Floris, looking at your commits in the **`qgisprocess`** repo I think you may have useful comments on this PR. Thanks for the inspiration!

Reply:
Hi @Robinlovelace it seems things are going well here :slightly_smiling_face:. Note that in a later commit I decided to install the extra plugins in a dedicated `extra_plugins` subdirectory, not straightly in the working directory where other folders may exist.

The line that you cite above refers to a line in `R-CMD-check.yaml`, but that line may not be up to date since that file evolves; copy the 'permanent link' to a code line or range in GitHub in order to link with a specific commit.

Reply:
Yes took me a while to work out where `extra_plugins` was created. Jannes already pointed to a folder in `/home/rstudio` in our image which we inherit from Rocker I believe. If you can give it a test and point out any issues, and take a look at `qgis/Dockerfile`, which is a bit messy sorry (should clean up!), that would be greatly appreciated. I think Jannes is looking at it now. I agree things are going well, more or less!

Reply:
I've not often used Docker, so would take too much now (currently also pushing healthy HD space limits, still without docker images). I'll invest in further preparing the package for CRAN. But let me know if you need hints when you get in trouble!

The alternative to the `extra_plugins` folder is to use the folder where QGIS normally downloads plugins, but I thought that might be less future-proof.